File: demo-wiremock/src/test/java/io/fries/wiremock/demo/wiremock/cucumber/world/WorldClock.java
package io.fries.wiremock.demo.wiremock.cucumber.world;

import org.springframework.stereotype.Component;

import java.time.ZonedDateTime;
import java.util.function.Supplier;

@Component
public class WorldClock {

    private Supplier<ZonedDateTime> clock;

    public WorldClock() {
        reset();
    }

    public final void reset() {
        clock = ZonedDateTime::now;
    }

    public void fixedAt(final ZonedDateTime zonedDateTime) {
        clock = () -> zonedDateTime;
    }

    public ZonedDateTime now() {
        return clock.get();
    }
}

File: examples/demo-cucumber/src/test/kotlin/io/fries/proxito/demo/cucumber/glue/world/WorldClock.kt
package io.fries.proxito.demo.cucumber.glue.world

import java.time.ZonedDateTime

class WorldClock(
    private var clock: () -> ZonedDateTime = { ZonedDateTime.now() }
) {

    fun fixedAt(zonedDateTime: ZonedDateTime) = apply {
        clock = { zonedDateTime }
    }

    fun now(): ZonedDateTime = clock.invoke()
}